emrun.py

```diff
@@ -540,16 +592,10 @@ def send_head(self):
 
     # A browser has navigated to this page - check which PID got spawned for
     # the browser
-    global previous_browser_process_pids, current_browser_process_pids, navigation_has_occurred
-    if not navigation_has_occurred and current_browser_process_pids is None:
-      running_browser_process_pids = list_processes_by_name(browser_exe)
-      for p in running_browser_process_pids:
-        def pid_existed(pid):
-          for proc in previous_browser_process_pids:
-            if proc['pid'] == pid:
-              return True
-          return False
-        current_browser_process_pids = list(filter(lambda x: not pid_existed(x['pid']), running_browser_process_pids))
+    global navigation_has_occurred, current_browser_processes
```

Comment:
I think you can trip current_browser_processes here since its not assigned to (`global` is only needed when thing are assigned to).

Reply:
That is true, though I find it is more documentative that way, when declaring one global for a set of related variables, to document the others as well. That makes the code more refactor proof. But dropped in any case.